usr/src/uts/armv8/os/fakebop.c

```diff
@@ -592,54 +970,485 @@ bop_panic(const char *fmt, ...)
 	bop_printf(NULL, "\nPress any key to reboot.\n");
 	while (BSVC_ISCHAR(SYSP) == 0) {}
 	bop_printf(NULL, "Resetting...\n");
+	for (;;) {}	/* XXXARM: spin forever, for now */
 }
 
-static int
-parse_value(char *p, uint64_t *retval)
+/*
+ * XXXARM: We should consider having bop_sysp again, which could
+ * call out to functions in inetboot/shim code during early boot.
+ */
+
+static void
+build_firmware_properties_fdt(const void *fdtp)
 {
-	int adjust = 0;
-	uint64_t tmp = 0;
-	int digit;
-	int radix = 10;
+	int property;
+	int node;
 
-	*retval = 0;
-	if (*p == '-' || *p == '~')
-		adjust = *p++;
+	node = fdt_subnode_offset(fdtp, 0, "chosen");
+	if (node < 0)
+		return;
 
-	if (*p == '0') {
-		++p;
-		if (*p == 0)
-			return (0);
-		if (*p == 'x' || *p == 'X') {
-			radix = 16;
-			++p;
+	fdt_for_each_property_offset(property, fdtp, node) {
+		const char *pn;
+		const void *pv;
+		int pl;
+
+		pv = fdt_getprop_by_offset(fdtp, property, &pn, &pl);
+		if (pv == NULL)
+			continue;
+		if (do_bsys_getproplen(NULL, pn) >= 0)
+			continue;
+
+		if (strcmp(pn, "phandle") == 0 ||
+		    strcmp(pn, "impl-arch-name") == 0 ||
+		    strcmp(pn, "mfg-name") == 0 ||
+		    strcmp(pn, "__bootpath") == 0) {
```

Comment:
I'm not sure `__bootpath`  actually gets used for anything, it could just be removed I think

Reply:
Yeah, it seems unused. The loaders are oddly insistent on setting it :-)